src/main/java/org/ethereum/vm/client/Repository.java

```diff
@@ -0,0 +1,166 @@
+
+import org.ethereum.vm.DataWord;
+
+public interface Repository {
+
+    /**
+     * Checks whether an account exists.
+     *
+     * @param address
+     *            the account address
+     * @return true if account exist, false otherwise
+     */
+    boolean isExist(byte[] address);
+
+    /**
+     * Creates an account if not exist.
+     *
+     * @param address
+     *            the account address
+     * @ImplNote trigger account creation
+     */
+    void createAccount(byte[] address);
+
+    /**
+     * Deletes an account.
+     *
+     * @param address
+     *            the account address
+     */
+    void delete(byte[] address);
+
+    /**
+     * Increases the account nonce of the given account by one.
+     *
+     * @param address
+     *            the account address
+     * @return new value of the nonce
+     * @ImplNote trigger account creation
+     */
+    long increaseNonce(byte[] address);
+
+    /**
+     * Sets the account nonce of the given account
+     *
+     * @param address
+     *            the account address
+     * @param nonce
+     *            new nonce
+     * @return new value of the nonce
+     * @ImplNote trigger account creation
+     */
+    long setNonce(byte[] address, long nonce);
+
+    /**
+     * Returns current nonce of a given account
+     *
+     * @param address
+     *            the account address
+     * @return value of the nonce
+     */
+    long getNonce(byte[] address);
+
+    /**
+     * Stores code associated with an account
+     *
+     * @param address
+     *            the account address
+     * @param code
+     *            that will be associated with this account
+     * @ImplNote trigger account creation
+     */
+    void saveCode(byte[] address, byte[] code);
+
+    /**
+     * Retrieves the code associated with an account
+     *
+     * @param address
+     *            the account address
+     * @return code in byte-array format, or NULL if not exist
+     */
+    byte[] getCode(byte[] address);
+
+    /**
+     * Puts a value in storage of an account at a given key
+     *
+     * @param address
+     *            the account address
+     * @param key
+     *            of the data to store
+     * @param value
+     *            is the data to store
+     * @ImplNote trigger account creation
+     */
+    void putStorageRow(byte[] address, DataWord key, DataWord value);
+
+    /**
+     * Retrieves storage value from an account for a given key
+     *
+     * @param address
+     *            the account address
+     * @param key
+     *            associated with this value
+     * @return the value, or NULL if not exist
+     */
+    DataWord getStorageRow(byte[] address, DataWord key);
+
+    /**
+     * Retrieves balance of an account
+     *
+     * @param address
+     *            the account address
+     * @return balance of the account as a <code>BigInteger</code> value
+     */
+    BigInteger getBalance(byte[] address);
```

Comment:
There's probably some discrepancy here.  We use long as our nano sem in Amount class.  There seems to be little point in using BigInt, as we do not support over longs

Reply:
This is because the base unit in Ethereum is 10^-18 ETH, while the base unit is 10^-9 SEM in Semux. Using Biginteger to handle any decimals.

Reply:
Ok, that could get tricky.  We'll probably want to add a toBigInt to amount class to make sure there's no errors.

Will need to figure out if this means we should move to 10^-18 in our db to match, or what.